### PR TITLE
config: support online change rocksdb config

### PR DESCRIPTION
**Pull request by NingLin-P**
This pr is part of online config change.
This pr add support for online change rocksdb config through config manager, under the hood, it is same as `Debugger::modify_tikv_config` except it only support change config on the `Config` struct.

###  What is the type of the changes?
- New feature (a change which adds functionality)

###  How is the PR tested?
- Integration test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?
No

###  Does this PR affect `tidb-ansible`?
No

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -712,10 +718,15 @@ pub struct DbConfig {
...
     pub enable_unordered_write: bool,
+    #[config(submodule)]
     pub defaultcf: DefaultCfConfig,
+    #[config(submodule)]
     pub writecf: WriteCfConfig,
+    #[config(submodule)]
     pub lockcf: LockCfConfig,
+    #[config(submodule)]
     pub raftcf: RaftCfConfig,
+    #[config(skip)]
```
Comment (Connor1996) [inline src/config.rs]: why not include Titan

File: src/config.rs [hunk 2]
```diff
@@ -1059,6 +1073,132 @@ impl RaftDbConfig {
...
+            _ => Err(format!("invalid cf {:?} for db {:?}", cf, self.db_type).into()),
+        }
+    }
+}
+
+impl ConfigManager for DBConfigManger {
+    fn dispatch(&mut self, change: ConfigChange) -> Result<(), Box<dyn Error>> {
+        let change_str = format!("{:?}", change);
+        let mut change: Vec<(String, ConfigValue)> = change.into_iter().collect();
+        let cf_config = change.drain_filter(|(name, _)| name.ends_with("cf"));
```
Comment (Connor1996) [inline src/config.rs]: what would the change string be like?

File: src/config.rs (hunk 1 above)
Comment (NingLin-P) [reply src/config.rs]: Do we now support change Titan config dynamically?

File: src/config.rs (hunk 2 above)
Comment (NingLin-P) [reply src/config.rs]: It will be the field name in the config struct like `defaultcf`.

File: src/config.rs [hunk 3]
```diff
@@ -1059,6 +1073,132 @@ impl RaftDbConfig {
...
+}
+
+fn config_to_slice(config_change: &[(String, String)]) -> Vec<(&str, &str)> {
+    config_change
+        .iter()
+        .map(|(name, value)| (name.as_str(), value.as_str()))
+        .collect()
+}
+
+// Convert `ConfigValue` to formated String that can pass to `DB::set_db_options`
```
Comment (rleungx) [inline src/config.rs]:
```suggestion
// Convert `ConfigValue` to formatted String that can pass to `DB::set_db_options`
```

File: src/config.rs (hunk 2 above)
Comment (NingLin-P) [reply src/config.rs]: The `change_str ` will be something like `{"rocksdb": {"defaultcf": {"target_file_size_base": 67108864b, "block_cache_size": 8388608b}, "max_background_jobs": 2}}`.

File: src/config.rs (hunk 1 above)
Comment (Connor1996) [reply src/config.rs]: `blob_run_mode` can be changed dynamically

**Pull request merged**